AwsEncryptionSDK/runtimes/net/Examples/Keyring/AwsKmsHierarchical/AwsKmsHierarchicalKeyring.cs

```diff
@@ -51,158 +58,121 @@
-        };
-        var hierarchicalKeyringA = materialProviders.CreateAwsKmsHierarchicalKeyring(createHierarchicalKeyringA);
+        // Demonstrate that the ciphertext and plaintext are different.
+        Assert.NotEqual(encryptOutput.Ciphertext.ToArray(), plaintext.ToArray());
         
-        var createHierarchicalKeyringB = new CreateAwsKmsHierarchicalKeyringInput()
+        var decryptInput = new DecryptInput
         {
-            KeyStore = keystore,
-            BranchKeyId = branchKeyIdB,
-            Cache = new CacheType { Default = new DefaultCache{EntryCapacity = 100} },
-            TtlSeconds = 600
+            Ciphertext = encryptOutput.Ciphertext,
+            EncryptionContext = encryptionContext,
+            Keyring = keyring
         };
-        var hierarchicalKeyringB = materialProviders.CreateAwsKmsHierarchicalKeyring(createHierarchicalKeyringB);
 
-        var decryptFailed = false;
-        try
-        {
-            // Try to use keyring for Tenant B to decrypt a message encrypted with Tenant A's key
-            // Expected to fail.
-            var decryptInput = new DecryptInput
-            {
-                Ciphertext = encryptOutputA.Ciphertext,
-                Keyring = hierarchicalKeyringB,
-            };
-            encryptionSDK.Decrypt(decryptInput);
-        }
-        catch (AwsCryptographicMaterialProvidersException)
-        {
-            decryptFailed = true;
-        }
-        Assert.True(decryptFailed);
+        // 7. Decrypt the Data
+        var decryptOutput = encryptionSdk.Decrypt(decryptInput);
         
-        decryptFailed = false;
-        try
-        {
-            // Try to use keyring for Tenant A to decrypt a message encrypted with Tenant B's key
-            // Expected to fail.
-            var decryptInput = new DecryptInput
-            {
-                Ciphertext = encryptOutputB.Ciphertext,
-                Keyring = hierarchicalKeyringA,
-            };
-            encryptionSDK.Decrypt(decryptInput);
-        }
-        catch (AwsCryptographicMaterialProvidersException)
-        {
-            decryptFailed = true;
-        }
-        Assert.True(decryptFailed);
-        
-        // Decrypt your encrypted data using the same keyring you used on encrypt.
-        encryptionSDK.Decrypt(new DecryptInput {
-            Ciphertext = encryptOutputA.Ciphertext,
-            Keyring = keyring }
-        );
-        encryptionSDK.Decrypt(new DecryptInput {
-            Ciphertext = encryptOutputB.Ciphertext,
-            Keyring = keyring }
-        );
-
+        // Demonstrate that the decrypted ciphertext and plaintext are the same
+        Assert.Equal(decryptOutput.Plaintext.ToArray(), plaintext.ToArray());
+        
+        // 8. Version the Branch Key in our KeyStore.
+        //    Only the branch key will be rotated. 
+        //    This rotation is eventually consistent. See the VersionBranchKey
+        //    Example for how to version a branch key.
```

Comment:
Issue: Can you describe Version as Rotate?  
It is implied here, but not explicitly said.